qa-tests-backend/src/test/groovy/ProcessesListeningOnPortsTest.groovy

```diff
@@ -228,6 +238,96 @@ class ProcessesListeningOnPortsTest extends BaseSpecification {
         assert list3.size() == 0
     }
 
+    @Tag("BAT")
+    def "Verify networking endpoints disappear when process is terminated"() {
+        given:
+        "Two deployments that listen on ports are started up"
+
+        rebuildForRetries()
+        def clusterId = ClusterService.getClusterId()
+
+        String deploymentId3 = targetDeployments[2].getDeploymentUid()
+
+        def gotCorrectNumElements = waitForResponseToHaveNumElements(1, deploymentId3, 240)
+
+        assert gotCorrectNumElements
+
+        def processesListeningOnPorts = evaluateWithRetry(10, 10) {
+                def temp = ProcessesListeningOnPortsService
+                        .getProcessesListeningOnPortsResponse(deploymentId3)
+                return temp
+        }
+
+        assert processesListeningOnPorts
+
+        def list = processesListeningOnPorts.listeningEndpointsList
+        assert list.size() == 1
+
+        def endpoint = list.find { it.endpoint.port == 8081 }
+
+        assert endpoint
+        assert endpoint.clusterId == clusterId
+        assert endpoint.containerName == TCPCONNECTIONTARGET3
+        assert endpoint.signal.id
+        assert endpoint.signal.containerId
+        assert endpoint.signal.time
+        assert endpoint.signal.name == "socat"
+        assert endpoint.signal.execFilePath == "/usr/bin/socat"
+        assert endpoint.signal.args == "-d -d -v TCP-LISTEN:8081,fork STDOUT"
+        assert endpoint.signal.pid
+
+        gotCorrectNumElements = waitForResponseToHaveNumElements(0, deploymentId3, 180)
```

Comment:
Checks that there are 0 listening endpoints returned by the API at a time when the listening endpoint is closed. Add a comment before merging.